staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go

```diff
@@ -96,24 +104,55 @@ type queueSet struct {
 	totRequestsExecuting int
 
 	emptyHandler fq.EmptyHandler
-	dealer       *shufflesharding.Dealer
 }
 
-// NewQueueSet creates a new QueueSet object.
-// There is a new QueueSet created for each priority level.
-func (qsf queueSetFactory) NewQueueSet(config fq.QueueSetConfig) (fq.QueueSet, error) {
-	fq := &queueSet{
-		clock:                qsf.clock,
-		counter:              qsf.counter,
-		estimatedServiceTime: 60,
-		config:               config,
-		lastRealTime:         qsf.clock.Now(),
+// NewQueueSetFactory creates a new QueueSetFactory object
+func NewQueueSetFactory(c clock.PassiveClock, counter counter.GoRoutineCounter) fq.QueueSetFactory {
+	return &queueSetFactory{
+		counter: counter,
+		clock:   c,
 	}
-	err := fq.SetConfiguration(config)
+}
+
+func (qsf *queueSetFactory) QualifyQueuingConfig(qCfg fq.QueuingConfig) (fq.QueueSetCompleter, error) {
+	dealer, err := checkConfig(qCfg)
 	if err != nil {
 		return nil, err
 	}
-	return fq, nil
+	return &queueSetCompleter{
+		factory: qsf,
+		qCfg:    qCfg,
+		dealer:  dealer}, nil
+}
+
+// checkConfig returns a non-nil Dealer if the config is valid and
+// calls for one, and returns a non-nil error if the given config is
+// invalid.
+func checkConfig(qCfg fq.QueuingConfig) (*shufflesharding.Dealer, error) {
+	if qCfg.DesiredNumQueues == 0 {
+		return nil, nil
+	}
+	dealer, err := shufflesharding.NewDealer(qCfg.DesiredNumQueues, qCfg.HandSize)
+	if err != nil {
+		err = errors.Wrap(err, "the QueueSetConfig implies an invalid shuffle sharding config (DesiredNumQueues is deckSize)")
```

Comment:
"DesiredNumQueues is deckSize" -- not sure I understand this.

Reply:
I expect it will make sense in the overal error message --- whose innermost part is in generic shuffle sharding terms (deckSize and handSize) and that is wrapped in remarks relating upward through layers of abstraction.